database/engine/metadata_log/logfile.c

```diff
@@ -273,18 +273,166 @@ static int check_metadata_logfile_superblock(uv_file file)
     assert(req.result >= 0);
     uv_fs_req_cleanup(&req);
 
-    if (strncmp(superblock->magic_number, RRDENG_METALOG_MAGIC, RRDENG_MAGIC_SZ) ||
-        strncmp(superblock->version, RRDENG_METALOG_VER, RRDENG_VER_SZ)) {
+    if (strncmp(superblock->magic_number, RRDENG_METALOG_MAGIC, RRDENG_MAGIC_SZ)) {
         error("File has invalid superblock.");
         ret = UV_EINVAL;
     } else {
         ret = 0;
     }
-    error:
+    if (strncmp(superblock->version, RRDENG_METALOG_VER, RRDENG_VER_SZ)) {
```

Comment:
This makes no sense as a way to store the version of the file structure. The versions are integers to allow simple comparison and to decide if the version on the disk is newer or older than the version in the code. Why is it being stored as a string inside a 16-byte array? This does not match the processing needs and simply adds a point of failure (the string is not an integer) that is not captured properly in this piece of code.

Reply:
Yes indeed - I wanted it to be an integer inside, not an ascii representation of an integer. Because it allows us to track cases like "all future versions" as `if (ver > CURRENT_VER)` and deal with them appropriately.

Reply:
You said you didn't care about ascii representation but it doesn't matter. I can do it binary and be done with it.